app/services/cosmos/cosmos.ts

```diff
@@ -85,26 +148,35 @@ export class CosmosAPI {
    * @param delegatorAddress The delegator address
    * @param validatorAddress The validator address
    */
-  async getDelegation(delegatorAddress: string, validatorAddress: string) {
-    return this.api.get.delegation(delegatorAddress, validatorAddress) as CosmosDelegation
+  async getDelegation(delegatorAddress: string, validatorAddress: string):
+    Promise<CosmosDelegation> {
+    const { delegationResponse } =
+      await this.queryClient.staking.delegation(delegatorAddress, validatorAddress);
+    return convertDelegationResponse(delegationResponse)
   }
 
   /**
    * Get all redelegations
    *
    * @param delegatorAddress The delegator address
    */
-  async getRedelegations(delegatorAddress: string) {
-    return this.api.get.redelegations(delegatorAddress) as CosmosDelegation[]
+  async getRedelegations(delegatorAddress: string):
+    Promise<CosmosRedelegation[]> {
+    const { redelegationResponses } =
+      await this.queryClient.staking.redelegations(delegatorAddress, 'sourceValidatorAddress', 'destinationValidatorAddress');
+    return redelegationResponses.map(res => convertRedelegation(res.redelegation))
   }
```

Comment:
Should we modify the input parameters to include `sourceValidatorAddress` and `destinationValidatorAddress`?

Reply:
I think the original functions wants to list all redelegation of a address, regard less of source/dst validation
Do we have similar api in cosmjs?

Reply:
Seems CosmJS only has this method about redelegation. Is there a way to customize a similar one by available functions? Say, get all the source/dst addresses first, then use `redelegations` method to list them? But we'll need to know if a delegation belongs to a redelegation.